src/metabase/search/filter.clj

```diff
@@ -256,10 +252,11 @@
   This is function instead of a def so that optional-filter-clause can be defined anywhere in the codebase."
   []
   (merge
-   ;; models support search-native-query if dataset_query is one of the searchable columns
-   {:search-native-query (->> (dissoc (methods search.config/searchable-columns-for-model) :default)
-                              (filter (fn [[k v]]
-                                        (contains? (set (v k)) :dataset_query)))
+   ;; models support search-native-query if there is additional columns to search when the `search-native-query`
+   ;; argument is true
+   {:search-native-query (->> (dissoc (methods search.config/searchable-columns) :default)
+                              (filter (fn [[model f]]
+                                        (seq (set/difference (set (f model true)) (set (f model false))))))
```

Comment:
Now we can decouple this function from the fact that `:dataset_query` is the only native query column

Reply:
When do you imagine us needing to extend the set of excluded columns?

Reply:
Unsure, maybe for notifications